### DISCUSSION TITLE
Binding broken

### DISCUSSION THREAD

**Discussion by filip-rakoczy-tfs** [yiszza/Material.Components.Maui#38] [General]
When binding a TextField to a Text property in my ViewModel I cannot get the text from the control to successfully propagate to the property. Text is always null. I'm using MVVM Community toolkit.

Add Command:
`    [RelayCommand]
    async Task Add()
    {
        if (string.IsNullOrWhiteSpace(Text))
            return;

        if (connectivity.NetworkAccess != NetworkAccess.Internet)
        {
            await Shell.Current.DisplayAlert("No Internet", "You need internet to add items", "OK");
            return;
        }

        Items.Add(Text);
        Text = string.Empty;
    }` 

text field:
    `[ObservableProperty]
    private string text;`

TextField compoment .xaml
`        <mdc:TextField Text="Please"
                       Grid.Row="1"
                       SupportingText="None"
                       LabelText="Add Task"/>`

**Comment by WongLord** [comment]
Try doing this:

text field:
`[ObservableProperty] public string text;`

TextField compoment .xaml
`<mdc:TextField Text="{Binding Text}" Grid.Row="1" SupportingText="None" LabelText="Add Task"/>`
